packages/legacy-client/src/client/actions/actions.ts

```diff
@@ -45,21 +45,37 @@ export interface ValidLegacyActionParameterTypes {
 export type ActionArgs<
   O extends OntologyDefinition<any>,
   A extends keyof O["actions"],
-> =
-  & {
+> = NonNullableKeys<O["actions"][A]["parameters"]> extends never ? {
```

Comment:
Basically, the problem was that our types as they were allowed you to pass in arrays to action args (based on our tests atleast). It was because if `NonNullableKeys` or `NullableKeys` were `never`, the type would look something like `{nullableKeys} & {}`, and for some reason having that `& {}` just allowed users to put in anything. Attempted to fix that here, but there may be a cleaner solution for this

Reply:
Specifically `{[P in never]: any}` accepts `[]`